internal/provider/cached_image_resource.go

```diff
@@ -293,6 +293,124 @@ func (r *CachedImageResource) Configure(ctx context.Context, req resource.Config
+	if !data.BaseImageCacheDir.IsNull() {
+		env["ENVBUILDER_BASE_IMAGE_CACHE_DIR"] = tfValueToString(data.BaseImageCacheDir)
+	}
+
+	if !data.BuildContextPath.IsNull() {
+		env["ENVBUILDER_BUILD_CONTEXT_PATH"] = tfValueToString(data.BuildContextPath)
+	}
+
+	if !data.CacheTTLDays.IsNull() {
+		env["ENVBUILDER_CACHE_TTL_DAYS"] = tfValueToString(data.CacheTTLDays)
+	}
+
+	if !data.DevcontainerDir.IsNull() {
+		env["ENVBUILDER_DEVCONTAINER_DIR"] = tfValueToString(data.DevcontainerDir)
+	}
+
+	if !data.DevcontainerJSONPath.IsNull() {
+		env["ENVBUILDER_DEVCONTAINER_JSON_PATH"] = tfValueToString(data.DevcontainerJSONPath)
+	}
+
+	if !data.DockerfilePath.IsNull() {
+		env["ENVBUILDER_DOCKERFILE_PATH"] = tfValueToString(data.DockerfilePath)
+	}
+
+	if !data.DockerConfigBase64.IsNull() {
+		env["ENVBUILDER_DOCKER_CONFIG_BASE64"] = tfValueToString(data.DockerConfigBase64)
+	}
+
+	if !data.ExitOnBuildFailure.IsNull() {
+		env["ENVBUILDER_EXIT_ON_BUILD_FAILURE"] = tfValueToString(data.ExitOnBuildFailure)
+	}
+
+	if !data.FallbackImage.IsNull() {
+		env["ENVBUILDER_FALLBACK_IMAGE"] = tfValueToString(data.FallbackImage)
+	}
+
+	if !data.GitCloneDepth.IsNull() {
+		env["ENVBUILDER_GIT_CLONE_DEPTH"] = tfValueToString(data.GitCloneDepth)
+	}
+
+	if !data.GitCloneSingleBranch.IsNull() {
+		env["ENVBUILDER_GIT_CLONE_SINGLE_BRANCH"] = tfValueToString(data.GitCloneSingleBranch)
+	}
+
+	if !data.GitHTTPProxyURL.IsNull() {
+		env["ENVBUILDER_GIT_HTTP_PROXY_URL"] = tfValueToString(data.GitHTTPProxyURL)
+	}
+
+	if !data.GitSSHPrivateKeyPath.IsNull() {
+		env["ENVBUILDER_GIT_SSH_PRIVATE_KEY_PATH"] = tfValueToString(data.GitSSHPrivateKeyPath)
+	}
+
+	if !data.GitUsername.IsNull() {
+		env["ENVBUILDER_GIT_USERNAME"] = tfValueToString(data.GitUsername)
+	}
+
+	if !data.GitPassword.IsNull() {
+		env["ENVBUILDER_GIT_PASSWORD"] = tfValueToString(data.GitPassword)
+	}
+
+	if !data.IgnorePaths.IsNull() {
+		env["ENVBUILDER_IGNORE_PATHS"] = strings.Join(tfListToStringSlice(data.IgnorePaths), ",")
+	}
+
+	if !data.Insecure.IsNull() {
+		env["ENVBUILDER_INSECURE"] = tfValueToString(data.Insecure)
+	}
+
+	// Default to remote build mode.
+	if data.RemoteRepoBuildMode.IsNull() {
+		env["ENVBUILDER_REMOTE_REPO_BUILD_MODE"] = "true"
+	} else {
+		env["ENVBUILDER_REMOTE_REPO_BUILD_MODE"] = tfValueToString(data.RemoteRepoBuildMode)
+	}
+
+	if !data.SSLCertBase64.IsNull() {
+		env["ENVBUILDER_SSL_CERT_BASE64"] = tfValueToString(data.SSLCertBase64)
+	}
+
+	if !data.Verbose.IsNull() {
+		env["ENVBUILDER_VERBOSE"] = tfValueToString(data.Verbose)
+	}
+
+	if !data.WorkspaceFolder.IsNull() {
+		env["ENVBUILDER_WORKSPACE_FOLDER"] = tfValueToString(data.WorkspaceFolder)
+	}
+
+	// Do ExtraEnv last so that it can override any other values.
+	var diag, ds diag.Diagnostics
+	if !data.ExtraEnv.IsNull() {
+		for key, elem := range data.ExtraEnv.Elements() {
+			if _, ok := env[key]; ok {
+				// This is a warning because it's possible that the user wants to override
+				// a value set by the provider.
```

Comment:
Tbh we could go either way here, provider overrides extra, extra overrides provider or error when both are set. I think this is a good approach as any 👍🏻

Reply:
Let's go with this approach for now and tweak as required. Erroring here seems a bit extreme and there may be legit use-cases that we just can't think of right now.